src/sagas/development.saga.js

```diff
@@ -0,0 +1,29 @@
+import { remote } from 'electron';
+import { call, put, takeEvery } from 'redux-saga/effects';
+import { SHOW_RESET_STATE_PROMPT, resetAllState } from '../actions';
+
+import ElectronStore from 'electron-store';
+
+const { dialog } = remote;
+
+export function* showResetDialog(): Saga<void> {
+  const response = yield call([dialog, dialog.showMessageBox], {
+    type: 'warning',
+    buttons: ['Reset', 'Cancel'],
+    defaultId: 1,
+    cancelId: 1,
+    title: `Reset state`,
+    message: `Are you sure you want to reset your application state?`,
+    detail: `All your imported & created projects are removed from the application.\nThey're still on your disk but you need to re-import them. Only use this if you're having problems with a broken application state`,
+  });
+
+  const confirmed = response === 0;
+  if (confirmed) {
+    window.electronStore.clear();

```

Comment:
I realized that actually, we can do this in a slightly more idiomatic way:

```js
import ElectronStore from 'electron-store';
const electronStore = new ElectronStore();
electronStore.clear();
```

It has the same effect (each instance still writes to the same backend), so we don't have to rely on global variables :D

Reply:
If we do this change, it'll break your test below, but actually I have a better way to handle that anyway.

Instead this should become:

```js
yield call([electronStore,electronStore.clear]);
```

This has exactly the same effect (it would be different if `window.electronStore.clear` was async, but since it's sync it's the same), except it means that we can test that the method was called without a mock. See details below.

Reply:
(I know the array syntax is weird, will check soon what the issue is with just `call(electronStore.clear)`)